### PR TITLE
Update staging client-go

### PR DESCRIPTION
**Pull request by mwielgus**
Executed commands:
```
git fetch --all
git merge upstream/master
godep restore
./hack/update-staging-client-go.sh
```
cc: @caesarxuchao @lavalamp

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36465)

### CODE REVIEW

File: staging/src/k8s.io/client-go/Godeps/Godeps.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
 	"ImportPath": "k8s.io/client-go",
-	"GoVersion": "go1.6",
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/Godeps/Godeps.json]: Can you manually revert GoVersion and GodepVersion changes?

**Comment by caesarxuchao** [comment]
Two comments, otherwise lgtm.

**Comment by caesarxuchao** [comment]
@mikedanese this PR is updating the staging area (/staging). There was no BUILD file in /staging, and this PR doesn't add any BUILD file to /staging. However, Jenkins reported the verify-bazel.sh failed. Do you know what we should do?

@justinsb had this problem in the morning as well.

File: staging/src/k8s.io/client-go/Godeps/Godeps.json (hunk 1 above)
Comment (mwielgus) [inline staging/src/k8s.io/client-go/Godeps/Godeps.json]: Done.

File: staging/src/k8s.io/client-go/LICENSE [hunk 2]
```diff
@@ -1,202 +0,0 @@
-
```
Comment (mwielgus) [inline staging/src/k8s.io/client-go/LICENSE]: Done.

**Comment by caesarxuchao** [comment]
@justinsb's PR passed the verification now: #35583.

**Comment by caesarxuchao** [comment]
@mwielgus I'm going to lgtm this PR. Could you add back the LICENSE file? If you didn't get to it before merging, could you add it back in the next PR?

**Comment by caesarxuchao** [comment]
@saad-ali this one only changes the staging/ area, so it doesn't affect the production code, could you approve it?

**Comment by saad-ali** [comment]
> @saad-ali this one only changes the staging/ area, so it doesn't affect the production code, could you approve it?

Ack. Thanks for the clarification @caesarxuchao. Approved for post-code freeze merge.

**Comment by mikedanese** [comment]
@k8s-bot unit test this

**Pull request merged**